github/resource_organization_block.go

```diff
@@ -0,0 +1,106 @@
+package github
+
+import (
+	"context"
+	"log"
+	"net/http"
+
+	"github.com/google/go-github/v25/github"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceOrganizationBlock() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceOrganizationBlockCreate,
+		Read:   resourceOrganizationBlockRead,
+		Delete: resourceOrganizationBlockDelete,
+		Importer: &schema.ResourceImporter{
+			State: func(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+				username := d.Id()
+				d.Set("username", username)
+				d.SetId(username)
+				return []*schema.ResourceData{d}, nil
+			},
+		},
+
+		Schema: map[string]*schema.Schema{
+			"username": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"etag": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceOrganizationBlockCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*Organization).client
+	orgName := meta.(*Organization).name
+	ctx := context.Background()
+	username := d.Get("username").(string)
+
+	log.Printf("[DEBUG] Creating organization block: %s (%s)", username, orgName)
+	_, err := client.Organizations.BlockUser(ctx, orgName, username)
+	if err != nil {
+		return err
+	}
+	d.SetId(username)
+
+	return resourceOrganizationBlockRead(d, meta)
+}
+
+func resourceOrganizationBlockRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*Organization).client
+	orgName := meta.(*Organization).name
+
+	username := d.Id()
+
+	ctx := context.WithValue(context.Background(), ctxId, d.Id())
+	if !d.IsNewResource() {
+		ctx = context.WithValue(ctx, ctxEtag, d.Get("etag").(string))
+	}
+
+	log.Printf("[DEBUG] Reading organization block: %s (%s)", d.Id(), orgName)
+	blocked, resp, err := client.Organizations.IsBlocked(ctx, orgName, username)
+	if err != nil {
+		if ghErr, ok := err.(*github.ErrorResponse); ok {
+			if ghErr.Response.StatusCode == http.StatusNotModified {
+				return nil
+			}
+			// not sure if this will ever be hit, I imagine just returns false?
+			if ghErr.Response.StatusCode == http.StatusNotFound {
+				log.Printf("[WARN] Removing organization block %s/%s from state because it no longer exists in GitHub",
+					orgName, d.Id())
+				d.SetId("")
+				return nil
+			}
+		}
+		return err
+	}
+
+	if !blocked {
+		d.SetId("")
+		return nil
+	}
+
+	d.Set("etag", resp.Header.Get("ETag"))
```

Comment:
Thoughts on adding `d.Set("username", username)`?

Reply:
I think that'll get around having to write custom import state https://github.com/terraform-providers/terraform-provider-github/pull/181/files